source/geometry_model/two_merged_boxes.cc

```diff
@@ -201,7 +201,11 @@ namespace aspect
       std::set< std::pair< std::pair<types::boundary_id, types::boundary_id>, unsigned int> > periodic_boundaries;
       for ( unsigned int i=0; i<dim+dim-1; ++i)
         if (periodic[i])
-          periodic_boundaries.insert( std::make_pair( std::pair<types::boundary_id, types::boundary_id>(2*i, 2*i+1), i) );
+          {
+            // The direction of periodicity (0,dim-1)
+            const unsigned int tmp_dim = i>=dim ? i-dim : i;
```

Comment:
This is the direction,  can you rename the variable to ``direction`` please? Then the comment is not needed anymore.